### ISSUE TITLE
inconsistent resources order returned by get-pipeline command

### ISSUE THREAD

**Issue by alpha-er** [concourse/concourse#2814]
fly version: 3.14.1

There will always be at least one team pipeline whose config when returned by `fly get-pipeline` command will have varying order of its resources. For example, first run of `fly get-pipeline -p mypipeline` returns:
```
...
resources:
- name: a
  type: git
- name: b
  type: docker-image
- name: c
  type: git
...
```
The next run may return:
```
...
resources:
- name: c
  type: git
- name: a
  type: git
- name: b
  type: docker-image
...
```
Affects concourse-pipeline-resource `check` command (concourse/concourse-pipeline-resource#38)

**Comment by edtan** [comment]
I noticed this in v4.2.2 after seeing thousands of pipeline versions.  I tried running the [resourcesQuery](https://github.com/concourse/concourse/blob/master/atc/db/resource.go#L60-L65) directly in postgres, and it looks like the rows are returned in different order from time to time (maybe every 10 seconds or so).

For testing, I used this [pipeline](https://gist.github.com/edtan/3cb44db45839fbab3ddb74f4c55602a0) and ran this [query](https://gist.github.com/edtan/89086d02e801af3792aa81635af070d1).

I'll do some more testing locally with adding an order by, and hopefully submit a PR later this week.
